tile_generator/templates/jobs/spec

```diff
@@ -112,9 +112,11 @@ properties:
 
 consumes:
 {% for release in context.releases.values() if release.consumes %}
-{% for link_type, link in release.consumes.iteritems() %}
-  - name: {{ link.name or link_type }}
-    type: {{ link_type }}
-    optional: {{ link.optional or true }}
+{% for link_name, link in release.consumes.iteritems() %}
+  - name: {{ link_name }}
+    type: {{ link.type or link_name }}
+{% if 'optional' in link and link.optional %}
```

Comment:
use the `if link.optional is defined` instead as pointed out here: http://jinja.pocoo.org/docs/2.10/templates/#builtin-tests and https://stackoverflow.com/questions/33400771/check-if-key-exists-in-a-dict-in-jinja2-template-on-ansible/33444226